sycl/source/detail/device_impl.cpp

```diff
@@ -435,6 +436,46 @@ std::string device_impl::getDeviceName() const {
   return MDeviceName;
 }
 
+/* On first call this function queries for device timestamp
+   along with host synchronized timestamp
+   and stores it in memeber varaible deviceTimePair.
+   Succive calls to this function would just retrieve the host timestamp ,
+   compute difference against the host timestamp in deviceTimePair
+   and calculate the device timestamp based on the difference.
+   deviceTimePair is refreshed with new device and host timestamp after a
+   certain interval (determined by timeTillRefresh) to account for clock skew
+   between host and device.
+*/
+
+uint64_t device_impl::getTime(){
+  constexpr uint64_t timeTillRefresh = 100e9;
+  uint64_t hostTime;
+  if (MIsHostDevice) {
+    using namespace std::chrono;
+    return duration_cast<nanoseconds>(steady_clock::now().time_since_epoch())
+        .count();
+  }
+  auto plugin = getPlugin();
+  RT::PiResult result =
+      plugin.call_nocheck<detail::PiApiKind::piGetDeviceAndHostTimer>(
+          MDevice, nullptr, &hostTime);
+  plugin.checkPiResult(result == PI_ERROR_INVALID_OPERATION ? PI_SUCCESS
+                                                            : result);
```

Comment:
Can't we call C++ API for getting current host time here? E.g. `duration_cast<nanoseconds>(steady_clock::now().time_since_epoch())      .count();`

Or we expect that the current host time returned by `piGetDeviceAndHostTimer`` may not correlate with "C++ host time`?

Reply:
Yes, I didn't want to the couple the host timer from `piGetDeviceAndHostTimer` to the C++ host timer, as a plugin may choose a different time base.

Reply:
I agree, but aligning everything with "C++ host timer" would allow users to compare events from different commands and backends, which might be useful.
Anyway, I do not insist, just trying to brainstorm.